lib/appcanary/configuration.rb

```diff
@@ -31,27 +59,78 @@ def configure(&block)
       block.call(configuration) if block
     end
 
+    # another way to do static configuration
+    def api_key=(val);      configuration.api_key = val;      end
+    def gemfile_lock=(val); configuration.gemfile_lock = val; end
+    def monitor_name=(val); configuration.monitor_name = val; end
+    def base_uri=(val);     configuration.base_uri = val;     end
+
+    # throws if config is broken
+    def check_runtime_config!(endpoint, config)
+      if endpoint == :monitors && config[:monitor_name].nil?
+        raise Appcanary::ConfigurationError.new("Appcanary.monitor_name = ???")
+      end
+    end
+
     def resolved_config
```

Comment:
Just spotted part 2 of this... I'm inclined to try both. `Dir.pwd` prints the current working directory for the process itself, so it would be contingent on the app within which the gem is being loaded not messing via `Dir.chdir` and the like. 

OTOH, I can't think of a non-bundler context for using this gem which *wouldn't* benefit from a quick look in the current directory for a usable file. Particularly for the yaml config file.

So I'm thinking, start with `Dir.pwd`, fall back to `Bundler.root`, and finally error out.